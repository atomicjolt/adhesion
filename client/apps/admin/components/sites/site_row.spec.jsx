import React from 'react';
import { shallow } from 'enzyme';
import SiteRow from './site_row';

describe('sites list row', () => {
  let result;
  let props;

  beforeEach(() => {
    props = {
      site: { url: 'http://www.example.com' },
    };
    result = shallow(<SiteRow {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles the first button onclick event', () => {
    expect(result.instance().state.siteModalOpen).toBeFalsy();
    result.find('button').at(0).simulate('click');
    expect(result.instance().state.siteModalOpen).toBeTruthy();
  });

  it('handles the second button onclick event', () => {
    const button = result.find('button');
    expect(result.instance().state.confirmDeleteModalOpen).toBeFalsy();
    button.at(1).simulate('click');
    expect(result.instance().state.confirmDeleteModalOpen).toBeTruthy();
  });
});
