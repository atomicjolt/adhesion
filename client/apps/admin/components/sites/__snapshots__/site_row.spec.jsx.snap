// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sites list row matches the snapshot 1`] = `ShallowWrapper {}`;
